src/api/catalog-search/catalog-search.ts

```diff
@@ -283,6 +285,27 @@ export class CatalogSearchAPI {
     }
   }
 
+  /**
+   * De-duplicates packages that appear multiple times, keeping the
+   * most recently published one.
+   */
+  private dedup(results: CatalogSearchResults): CatalogSearchResults {
+    const dedupedResults: Map<string, ExtendedCatalogPackage> = new Map();
+
+    results.forEach((pkg) => {
```

Comment:
Style nit: prefer `for-if` instead of `forEach()`. It's easier to read and maintain in the long term. For example, it is not possible to `break` a `forEach()` loop.